src/protocol_nmea.cpp

```diff
@@ -2190,6 +2148,50 @@ uint32_t nmea_parse_ascb(int pHandle, const char msg[], int msgSize, rmci_t rmci
 uint32_t nmea_parse_asce(int pHandle, const char msg[], int msgSize, rmci_t rmci[NUM_COM_PORTS])
 {
 	(void)msgSize;
+
+	if(pHandle >= NUM_COM_PORTS)
+	{
+		return 0;
+	}
+	char *ptr = (char *)&msg[6];				// $ASCE
```

Comment:
I'm getting a little knit-picky here, but this is a bit messy with variable definitions mixed with conditions, etc mixed with curious white spacing choices, etc.  I think this, and the following lines of the rest of this function could probably benefit from some inline comments as well, to better communicate what its happening...

With the apparently non-trivial reworking of this (and the subsequent) functions, let's go ahead and add a doxygen block to these two functions as well.